### PR TITLE
hugo: init at 0.16

### PR DESCRIPTION
**Pull request by schneefux**
###### Motivation for this change

Added the [hugo](https://gohugo.io) static site generator with its dependencies.
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [x] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by schneefux** [comment]
I rebased on master, Travis and nox-review succeed now.
The same goes for #16300.

**Comment by joachifm** [comment]
Needs rebase

**Comment by schneefux** [comment]
Rebased.

**Comment by siddharthist** [comment]
@schneefux Needs rebase again :/

File: pkgs/applications/misc/hugo/deps.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    "packages": [
+        "gopkg.in/yaml.v2",
+        "github.com/hashicorp/hcl",
+        "github.com/pkg/sftp",
+        "golang.org/x/sys",
+        "github.com/hashicorp/go-multierror",
+        "golang.org/x/crypto",
+        "github.com/pkg/errors",
+        "github.com/kr/fs",
+	"github.com/kyokomi/emoji",
```
Comment (grahamc) [inline pkgs/applications/misc/hugo/deps.json]: There is some inconsistency in the indentation here.

Comment (siddharthist) [inline pkgs/applications/misc/hugo/deps.json]: Inconsistent indentation here

**Comment by fpletz** [comment]
Fixed the merge conflict & whitespace issues. Pushed to master as 4463f9f3d461615601b0280740f550e96026d129.

**Pull request closed without merging**